src/PatternBuilder.ts

```diff
@@ -47,116 +47,174 @@ type TermSequence = { tag: 'termSequence'; terms: ReadonlyArray<Term> }
-    }
-    const [c1, c2] = range
-    const lower = typeof c1 === 'string' ? c1.charCodeAt(0) : c1
-    const upper = typeof c2 === 'string' ? c2.charCodeAt(0) : c2
-    return { lower, upper } as const
-  }),
+  ranges: convertRanges(ranges),
 })
 
-/** @since 1.0.0 */
-export const subgroup = (subpattern: Pattern): Atom => ({
+/**
+ * Turn a `Pattern` into an `Atom`. In regular expression terms, this is wrapping the
+ * pattern in parentheses.
+ *
+ * @since 1.0.0
+ */
+export const subgroup: (subpattern: Pattern) => Atom = subpattern => ({
   tag: 'atom',
   kind: 'subgroup',
   subpattern,
 })
 
-/** @since 1.0.0 */
-export const anyNumber =
-  (opts: { greedy: boolean } = { greedy: false }) =>
-  (atom: Atom): QuantifiedAtom => ({
+/**
+ * Repeat an `Atom` any number of times (including zero).
+ *
+ * @since 1.0.0
+ */
+export const anyNumber: (opts?: { greedy: boolean }) => (atom: Atom) => QuantifiedAtom =
+  (opts = { greedy: false }) =>
+  atom => ({
     tag: 'quantifiedAtom',
     atom,
     greedy: opts.greedy,
     kind: 'star',
   })
 
-/** @since 1.0.0 */
-export const atLeastOne =
-  (opts: { greedy: boolean } = { greedy: false }) =>
-  (atom: Atom): QuantifiedAtom => ({
+/**
+ * Repeat an `Atom` any number of times, but at least once.
+ *
+ * @since 1.0.0
+ */
+export const atLeastOne: (opts?: { greedy: boolean }) => (atom: Atom) => QuantifiedAtom =
+  (opts = { greedy: false }) =>
+  atom => ({
     tag: 'quantifiedAtom',
     atom,
     greedy: opts.greedy,
     kind: 'plus',
   })
 
-/** @since 1.0.0 */
-export const maybe = (atom: Atom): QuantifiedAtom => ({
+/**
+ * Make an `Atom` optional -- it can occur in the pattern once or not at all.
+ *
+ * @since 1.0.0
+ */
+export const maybe: (atom: Atom) => QuantifiedAtom = atom => ({
   tag: 'quantifiedAtom',
   atom,
   greedy: false,
   kind: 'question',
 })
 
-/** @since 1.0.0 */
-export const times =
-  (count: number) =>
-  (atom: Atom): QuantifiedAtom => ({
+/**
+ * Repeat an `Atom` an exact number of times. (Aliased to `exactly` for better readability
+ * in some situations)
+ *
+ * @since 1.0.0
+ */
+export const times: (count: number) => (atom: Atom) => QuantifiedAtom =
+  count => atom => ({
     tag: 'quantifiedAtom',
     atom,
     greedy: true,
     kind: 'exactly',
     count,
   })
 
-/** @since 1.0.0 */
-export const atLeast =
-  (min: number) =>
-  (atom: Atom): QuantifiedAtom => ({
-    tag: 'quantifiedAtom',
-    atom,
-    greedy: true,
-    kind: 'minimum',
-    min,
-  })
+/**
+ * Alias of `times`
+ *
+ * @since 1.0.0
+ */
+export const exactly: (count: number) => (atom: Atom) => QuantifiedAtom = times
 
-/** @since 1.0.0 */
-export const between =
-  (min: number, max: number) =>
-  (atom: Atom): QuantifiedAtom => ({
+/**
+ * Repeat an `Atom` at least some number of times. For example, `atLeast(3)(char('a'))`
+ * represents `aaa`, `aaaaaa`, and `aaaaaaaaaaaaaaaaaaaaaaaa` but not `aa`
```

Comment:
https://youtu.be/yBLdQ1a4-JI

Reply:
not this one though https://www.youtube.com/watch?v=yBLdQ1a4-JI&t=37s